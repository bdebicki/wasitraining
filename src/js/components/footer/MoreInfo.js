import LINK from '../../elementHandlers/link';
import FOOTER from './elementHandlers/footer';
import { COPYRIGHTS } from '../../enums/content';
import addElement from '../../utils/addElement';

export default class MoreInfo {
	static showAboutDetails(e) {
		e.preventDefault();

		console.log('about dialogbox'); // eslint-disable-line no-console
	}

	static renderCopyrights() {
		return addElement('p', {
			children: COPYRIGHTS,
			classNames: FOOTER.COPYRIGHTS_CLASS,
		});
	}

	static renderAboutLink() {
		return addElement('a', {
			children: 'about',
			classNames: [
				LINK.BASIC_CLASS,
				LINK.INVERTED_STYLE_CLASS,
				FOOTER.ABOUT_LINK_CLASS,
			],
			href: '#about',
			onClick: MoreInfo.showAboutDetails,
		});
	}

	static render() {
		return addElement('section', {
			children: [
				MoreInfo.renderCopyrights(),
				MoreInfo.renderAboutLink(),
			],
			classNames: FOOTER.ABOUT_SECTION_CLASS,
		});
	}
}
