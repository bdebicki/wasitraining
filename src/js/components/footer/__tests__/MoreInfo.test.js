import MoreInfo from '../MoreInfo';

describe('test more info section', () => {
  it('render more info section', () => {
    const more = MoreInfo.render();

    expect(more).toMatchSnapshot();
  });
});
