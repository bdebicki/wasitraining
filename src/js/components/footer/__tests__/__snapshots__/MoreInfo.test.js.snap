// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test more info section render more info section 1`] = `
<section
  class="aboutSection"
>
  <p
    class="aboutSection__copyrights"
  >
    made with ‚ù§ in Gdynia 2017
  </p>
  <a
    class="link link--inverted aboutSection__aboutLink"
    href="#about"
  >
    about
  </a>
</section>
`;
